src/rgen/Microsoft.Macios.Transformer/DataModel/Method.Transformer.cs

```diff
@@ -94,6 +94,15 @@ public static bool TryCreate (MethodDeclarationSyntax declaration, RootContext c
 	/// </summary>
 	/// <returns>A constructor data model if the method is one in the bindings or null otherwise.</returns>
 	public Constructor? ToConstructor ()
-		=> IsConstructor ? new Constructor (this) : null;
+	{
+		if (!IsConstructor)
+			return null;
+
+		// do not allow constructors to have ref parameters
+		if (Parameters.Any (p => p.ReferenceKind != ReferenceKind.None))
+			return null;
```

Comment:
If the user tries to do something like this, and we don't support it, then IMHO it's better to show an error than to silently ignore it.

Reply:
This is of the transformation from old sdk to new, C# supports it. But I don't know how we would deal a constructor that add an 'out NSError'. 

Lets close the PR, is small and simple to reopen if needed.

Reply:
A constructor with 'out NSError' is a bad idea, but unfortunately we have a few of those.

We have a test to make sure we don't add more:

https://github.com/xamarin/xamarin-macios/blob/af7adf411e62ae966082f62a9a6ee201aed4db78/tests/cecil-tests/ConstructorTest.cs#L522-L557

and here's the list of current known failures for that test:

https://github.com/xamarin/xamarin-macios/blob/af7adf411e62ae966082f62a9a6ee201aed4db78/tests/cecil-tests/ConstructorTest.KnownFailures.cs#L81-L154